lib/services/elasticsearch.js

```diff
@@ -153,665 +173,682 @@ class ElasticSearch extends Service {
    *
    * @param {Request} request
-   * @returns {Promise}
+   * @return {Promise}
    */
-  createCollection(request) {
-    const esRequest = initESRequest(request);
+  getMapping (index, collection, { includeKuzzleMeta=false } = {}) {
+    const
+      esIndex = this._getESIndex(index, collection),
+      esRequest = {
+        index: esIndex
+      };
 
-    return this.kuzzle.indexCache.exists(esRequest.index)
-      .then(indexExists => {
-        if (! indexExists) {
-          this.throw('index_does_not_exist');
+    return this.kuzzle.indexCache.exists(index, collection)
+      .then(exists => {
+        if (! exists) {
+          this.throw('index_or_collection_does_not_exist', index, collection);
         }
 
-        return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type);
+        return this.client.indices.getMapping(esRequest);
       })
-      .then(collectionExists => {
-        if (collectionExists) {
-          return this.updateMapping(request);
-        }
-
-        const requestBody = request.input.body || {};
-
-        this._checkMapping(requestBody);
-
-        esRequest.body = this._mergeDefaultMapping(
-          esRequest.index,
-          {},
-          requestBody
-        );
-
-        return this.client.indices.putMapping(esRequest)
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
+      .then(({ body }) => {
+        const
+          properties = includeKuzzleMeta
+            ? body[esIndex].mappings.properties
+            : _.omit(body[esIndex].mappings.properties, '_kuzzle_info'),
+          apiResponse = {
+            dynamic: body[esIndex].mappings.dynamic,
+            _meta: body[esIndex].mappings._meta,
+            properties
+          };
+
+        return apiResponse;
       });
   }
 
   /**
    * Empty the content of a collection. Keep the existing mapping.
    *
-   * @param {Request} request
-   * @returns {Promise}
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   *
+   * @returns {Promise} { }
    */
-  truncateCollection(request) {
-    const
-      deleteRequest = new Request({
-        index: request.input.resource.index,
-        collection: request.input.resource.collection,
-        body: {
-          query: {
-            match_all: {}
-          }
-        }
-      }, {
-        user: request.context.user
-      });
+  truncateCollection (index, collection) {
+    let mappings;
 
-    deleteRequest.input.args.refresh = request.input.args.refresh || false;
+    const esRequest = {
+      index: this._getESIndex(index, collection)
+    };
+
+    return this.getMapping(index, collection)
+      .then(collectionMappings => {
+        mappings = collectionMappings;
+
+        return this.client.indices.delete(esRequest);
+      })
+      .then(() => this.client.indices.create({ ...esRequest, mappings }))
+      .then(() => {
+        const apiResponse = {};
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)